### ISSUE TITLE
SAML configuration updates after the initial configuration is set do not 'take'

### ISSUE THREAD

**Issue by uberbrady** [grokability/snipe-it#8852]
A customer re-issued a certificate for their SAML app, and then tried to update the configuration within Snipe-IT. The updated certificate and key would not reflect in the database (or in the web UI), no matter what I did. I even tried to paste in the raw XML configuration - but the old data persisted.

We need to be able to blank those out, or, even better, detect changes and correctly invalidate the database-backed certificate and key. The tricky part is that if you specify a metadata URL, the actual *data* at that URL can change, but the URL itself might *not* change - and Laravel may see that as a no-op.

One easy way to handle this might be to *always* invalidate the certificate and key on Save - even if nothing seems to have changed? Though I would fear that if the Metadata URL were to somehow go offline temporarily, hitting 'save' on the SAML settings might blank out your currently-working SAML config. Though, if you don't want to modify your SAML settings, then don't hit 'save' - that's probably kind of obvious?

**Comment by johnson-yi** [comment]
Hi uberbrady

The key pair (saml_sp_x509cert and saml_sp_privatekey column) in the database belongs to Snipe-IT. It only gets generated once on enabling SAML if there is no existing key pair. This is mainly used for Single Log Out where most IdPs require the LogoutRequest to be signed by the SP. 

The SAML IdP Metadata contains information for IdP entity id, SSO endpoint, SLO endpoint (if available) and IdP public certificate. This XML file would contain all the information required for SAML SSO from the Identity Provider. In your case updating the metadata should have resolved your issue.

The metadata is only checked to be valid when saving in SAML settings. This metadata is actually used and cached to file indefinitely at the next time you reach the login screen. This cached file should be getting invalidated any time you change/save any Snipe-IT settings (it compares its saved updated_at value against the updated_at column in settings).

**Comment by uberbrady** [comment]
I'm still not 100% convinced that I was doing it right - I **did** start to troubleshoot the customers XML, and checking keys, and certs. I would've figured the 'save' would've updated it, but I was frantically trying to get the customer's SSO working again, and I may not have been as careful as I should've been. I'll play around with it and confirm that it's really a bug before I get too mired into this.

**Comment by uberbrady** [comment]
I think it's possible that the issue was something *else* within the Settings object - perhaps the new requirement for having a stronger minimum password. We eventually fixed that in a migration (yanking customers minimum password length requirement up from whatever it was), so I think this isn't a 'real' issue. I also haven't seen it come up again, and I've worked with many, many customers on their SAML/SSO setups.
